demisto_sdk/commands/common/hook_validations/correlation_rule.py

```diff
@@ -84,3 +85,18 @@ def is_files_naming_correct(self):
                 self._is_valid = False
                 return False
         return True
+
+    @error_codes("CR102")
+    def validate_execution_mode_search_window(self):
+        """
+        Validates 'search_window' existence and non-emptiness for 'execution_mode' = 'SCHEDULED'.
+        """
+        if ("search_window" not in self.current_file) or (
+            self.current_file["execution_mode"] == "SCHEDULED"
+            and not self.current_file["search_window"]
+        ):
+            error_message, error_code = Errors.correlation_rules_execution_mode_error()
+            if self.handle_error(error_message, error_code, file_path=self.file_path):
+                self._is_valid = False
```

Comment:
do we usually change `_is_valid`? Shouldn't it be modified by whatever method calls this one?